connector/connect/server/src/main/scala/org/apache/spark/sql/connect/ml/MLHandler.scala

```diff
@@ -0,0 +1,162 @@
+                                   val method: Option[String],
+                                   val argValues: Array[Object] = Array.empty,
+                                   val argClasses: Array[Class[_]] = Array.empty) {
+
+  val methodChain = method.map(n => s"$objIdentifier.$n").getOrElse(objIdentifier)
+  private val methodChains = methodChain.split("\\.")
+  private val modelId = methodChains.head
+
+  private lazy val model = sessionHolder.mlCache.get(modelId)
+  private lazy val methods = methodChains.slice(1, methodChains.length)
+
+  def getAttribute: Any = {
+    assert(methods.length >= 1)
+    if (argValues.length == 0) {
+      methods.foldLeft(model.asInstanceOf[Object]) { (obj, attribute) =>
+        invokeMethod(obj, attribute)
+      }
+    } else {
+      val lastMethod = methods.last
+      if (methods.length == 1) {
+        invokeMethod(model.asInstanceOf[Object], lastMethod, argValues, argClasses)
+      } else {
+        val prevMethods = methods.slice(0, methods.length - 1)
+        val finalObj = prevMethods.foldLeft(model.asInstanceOf[Object]) { (obj, attribute) =>
+          invokeMethod(obj, attribute)
+        }
+        invokeMethod(finalObj, lastMethod, argValues, argValues)
+      }
+    }
+  }
+
+  def transform(modelRelation: proto.MlRelation.ModelTransform): DataFrame = {
+    // Create a copied model to avoid concurrently modify model params.
+    val copiedModel = model.copy(ParamMap.empty).asInstanceOf[Model[_]]
+    MLUtils.setInstanceParams(copiedModel, modelRelation.getParams)
+    val inputDF = MLUtils.parseRelationProto(modelRelation.getInput, sessionHolder)
+    copiedModel.transform(inputDF)
+  }
+}
+
+private object ModelAttributeHelper {
+  def apply(sessionHolder: SessionHolder,
+            modelId: String,
+            methodChain: Option[String] = None,
+            args: Array[proto.FetchModelAttr.Args] = Array.empty): ModelAttributeHelper = {
+    val tmp = deserializeMethodArguments(args, sessionHolder)
+    val argValues = tmp.map(_._1)
+    val argClasses = tmp.map(_._2)
+    new ModelAttributeHelper(sessionHolder, modelId, methodChain, argValues, argClasses)
+  }
+}
+
+object MLHandler extends Logging {
+  def handleMlCommand(
+      sessionHolder: SessionHolder,
+      mlCommand: proto.MlCommand): proto.MlCommandResponse = {
+
+    val mlCache = sessionHolder.mlCache
+
+    mlCommand.getMlCommandTypeCase match {
+      case MlCommandTypeCase.FIT =>
+        val fitCmd = mlCommand.getFit
+        val estimatorProto = fitCmd.getEstimator
+        assert(estimatorProto.getType == proto.MlStage.StageType.ESTIMATOR)
+
+        val name = fitCmd.getEstimator.getName
+        val params = fitCmd.getEstimator.getParams
+        val dataset = MLUtils.parseRelationProto(fitCmd.getDataset, sessionHolder)
+
+        val estimatorName = name.replace("pyspark", "org.apache.spark")
+        val estimator = MLUtils.getEstimator(estimatorName)
```

Comment:
please move name replacement to MLUtils.getEstimator(), where we will support custom implementations

Reply:
Add todo here for plugin